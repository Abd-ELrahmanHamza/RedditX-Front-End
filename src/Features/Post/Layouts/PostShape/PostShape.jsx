import VotingBar from "Features/Post/Components/VotingBar/VotingBar";
import PostBar from "../PostBar/PostBar";


import {Container} from "./PostShape.styled"

/**
 * PostShape Component 
 * @returns {React.Component} PostShape component
 */
const PostShape = ({fullPost}) => {
  return (
    <Container fullPost={fullPost}>
      <VotingBar number={2004}/>
      <PostBar fullPost={fullPost}/>
    </Container>
  );
};

export default PostShape;
