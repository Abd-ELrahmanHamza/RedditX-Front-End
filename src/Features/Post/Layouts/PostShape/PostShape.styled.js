import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  background-color: #cccccc2b;
  height: auto;
  width: 650px;
  max-width: 100%;
  margin: 0 auto;
  border-radius: 5px;
  cursor: pointer;
  border: 1px solid transparent;

  &:hover {
    border-color: black !important;
  }
`;

export const PostBar = styled.div`
  width: 95%;
  heigth: 100%;
  background-color: red;
`;
