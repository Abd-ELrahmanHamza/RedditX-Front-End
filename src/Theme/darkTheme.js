/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
  },
  // The colors of the backgrounds in the website
  background: {
    primary: "#1a1a1b",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
  },
};

export default darkTheme;
