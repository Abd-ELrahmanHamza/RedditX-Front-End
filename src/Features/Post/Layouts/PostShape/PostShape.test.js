import { shallow } from "enzyme";
import { screen, render, waitFor } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import darkTheme from "Theme/darkTheme";
import PostShape from "./PostShape";
import { BrowserRouter } from "react-router-dom";

describe("PostShape Component", () => {
  const wrapper = shallow(<PostShape />);

  it("this is a test for PostShape Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
