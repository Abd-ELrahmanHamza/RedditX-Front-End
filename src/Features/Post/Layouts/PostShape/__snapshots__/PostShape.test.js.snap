// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostShape Component this is a test for PostShape Component 1`] = `
<styled.div>
  <VotingBar
    number={2004}
  />
  <PostBar />
</styled.div>
`;
